src/main/scala/io/iohk/ethereum/nodebuilder/tooling/StorageConsistencyChecker.scala

```diff
@@ -0,0 +1,29 @@
+package io.iohk.ethereum.nodebuilder.tooling
+
+import com.typesafe.scalalogging.Logger
+
+import io.iohk.ethereum.db.storage.BlockHeadersStorage
+import io.iohk.ethereum.db.storage.BlockNumberMappingStorage
+
+object StorageConsistencyChecker {
+  type ShutdownOp = () => Unit
+
+  val DefaultStep = 1000
+
+  def checkStorageConsistency(
+      bestBlockNumber: BigInt,
+      blockNumberMappingStorage: BlockNumberMappingStorage,
+      blockHeadersStorage: BlockHeadersStorage,
+      shutdown: ShutdownOp,
+      step: Int = DefaultStep
+  )(implicit log: Logger): Unit =
+    Range(0, bestBlockNumber.intValue, step).foreach { idx =>
```

Comment:
I'm wondering if it makes sense to check every 1000th block. I imagine a case where the check passes but the db is inconsistent.
Perhaps it would be worthwhile to occasionally check the whole range?

Reply:
this is a first pass. I want to start identifying the circumstances when clients loose consistency.